test/connector/tcp/mssql/mssql_connector_test.go

```diff
@@ -50,6 +50,42 @@ func TestMSSQLConnector(t *testing.T) {
 		assert.EqualValues(t, testString, actualString)
 	})
 
+	t.Run("Passes valid database name to MSSQL through Secretless", func(t *testing.T) {
+		port := 2223
+		server := testutil.SecretlessHost
+
+		// Open the connection
+		connString := fmt.Sprintf("server=%s;port=%d;encrypt=disable;database=master", server, port)
+		conn, err := sql.Open("mssql", connString)
+		assert.Nil(t, err, "you can open the db connection")
+		if err != nil {
+			return
+		}
+		defer conn.Close()
+
+		// Ping the DB
+		err = conn.Ping()
+		assert.NoError(t, err, "valid db should not error")
+	})
+
+	t.Run("Passes invalid database name to MSSQL through Secretless", func(t *testing.T) {
+		port := 2223
+		server := testutil.SecretlessHost
+
+		// Open the connection
+		connString := fmt.Sprintf("server=%s;port=%d;encrypt=disable;database=meow", server, port)
+		conn, err := sql.Open("mssql", connString)
+		assert.Nil(t, err, "you can open the db connection")
+		if err != nil {
+			return
+		}
+		defer conn.Close()
+
+		// Ping the DB
+		err = conn.Ping()
+		assert.Error(t, err, "invalid db should error")
+	})
+
```

Comment:
Was thinking about this a bit more... both of these feel more like integration tests that we should be doing through `sqlcmd`.  They're really not unit tests.  There are no doubles or mocks.  We're not testing our application logic directly.  We're testing that the whole thing works end to end.

Reply:
@jonahx 

sqlcmd and go-mssqldb are equivalent clients. The idea of the fork is for it to remain the same as the original but behave differently iff Secretless calls it. In the test cases the client is being used in its original form. You're right that we might inadvertently introduce changes to the fork that might influence the tests but I think the probability is quite low. That would be a massive bug. 

If anything we ought to run the tests with both clients... there've been bugs that have show themselves only via one client.

You're also right that switching to `sqlcmd` tets wouldn't be much work after that PR is mergedd.

Reply:
Summary of offline discussion where we hammered this out:

1. `sqlcmd` will be used for our canonical tests, since that's the official MS client and is the most reliable, independent verification that our code works.
2. Kumbi will try to use the (unforked) denis repo with his existing tests, so that we have cross validation with 2 clients.

Reply:
Worth noting 2. could be pushed to a followon issue that could be completed any time...